packages/datadog-instrumentations/src/dns.js

```diff
@@ -67,28 +68,31 @@ function wrap (prefix, fn, expectedArgs, rrtype) {
       startArgs.push(rrtype)
     }
 
-    const asyncResource = new AsyncResource('bound-anonymous-fn')
-    return asyncResource.runInAsyncScope(() => {
-      startCh.publish(startArgs)
+    const context = { args: startArgs }
+    startCh.publish(context)
 
-      arguments[arguments.length - 1] = asyncResource.bind(function (error, result) {
-        if (error) {
-          errorCh.publish(error)
-        }
-        finishCh.publish(result)
-        cb.apply(this, arguments)
-      })
+    arguments[arguments.length - 1] = function (error, result) {
+      if (error) {
+        context.error = error
+        errorCh.publish(context)
+      }
+      context.result = result
+      asyncEndCh.publish(context)
+      cb.apply(this, arguments)
```

Comment:
As discussed offline, let's make sure this is wrapped between 2 events so that the subscriber can restore the right context only in the callback, similar to promises and to what is described in https://github.com/nodejs/node/pull/44943#issuecomment-1283503006.